pkg/icingaredis/telemetry/heartbeat.go

```diff
@@ -0,0 +1,102 @@
+package telemetry
+
+import (
+	"context"
+	"github.com/go-redis/redis/v8"
+	"github.com/icinga/icingadb/internal"
+	"github.com/icinga/icingadb/pkg/com"
+	"github.com/icinga/icingadb/pkg/icingaredis"
+	"github.com/icinga/icingadb/pkg/logging"
+	"github.com/icinga/icingadb/pkg/periodic"
+	"github.com/icinga/icingadb/pkg/types"
+	"github.com/icinga/icingadb/pkg/utils"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+	"strconv"
+	"time"
+)
+
+// StartHeartbeat periodically writes heartbeats to Redis for being monitored by Icinga 2.
+func StartHeartbeat(
+	ctx context.Context, client *icingaredis.Client, logger *logging.Logger,
+	instanceId types.Binary, envId *com.Atomic[types.Binary], db *sqlx.DB, dbErr *com.Atomic[error],
+) {
+	type ha struct {
+		isSet bool
+
+		active, inactive, weAreActive uint8
+	}
+
+	haInfo := &com.Atomic[ha]{}
+
+	periodic.Start(ctx, time.Second, func(_ periodic.Tick) {
+		values := map[string]string{"version": internal.Version}
+		if err, _ := dbErr.Load(); err != nil {
+			values["db-err"] = err.Error()
+		}
+
+		if ha, _ := haInfo.Load(); ha.isSet {
+			values["ha:active"] = strconv.FormatUint(uint64(ha.active), 10)
+			values["ha:inactive"] = strconv.FormatUint(uint64(ha.inactive), 10)
+			values["ha:we-are-active"] = strconv.FormatUint(uint64(ha.weAreActive), 10)
```

Comment:
Those names cloud be better. The types of each value is `uint8`, however the last one is actually used as a boolean. Typically, the other ones will be 0 or 1 in most situations making is easy to mistake them for booleans as well. Maybe something like `ha:active` for this instance and `ha:num-other-(in)active` for the other values (or maybe these values change completely depending on the result of my other comment).

Reply:
Cloud☁️, huh? :)